tests/pos/i1503.scala

```diff
@@ -0,0 +1,6 @@
+object Test {
+  (
+new Function0[Unit] {
+    def apply() = println("hello")
```

Comment:
Do we really need a separate pos test since the same code will be compiled by the run test?